packages/ec-core/src/operator/mutator/mod.rs

```diff
@@ -26,8 +209,117 @@ where
     type Output = G;
     type Error = anyhow::Error;
 
+    /// Apply this `Mutator` as an `Operator`
     fn apply(&self, genome: G, rng: &mut ThreadRng) -> Result<Self::Output, Self::Error> {
         self.mutator.mutate(genome, rng)
     }
 }
 impl<M> Composable for Mutate<M> {}
+
+/// Implement [`Mutator`] for a reference to a [`Mutator`].
+/// This allows us to wrap a reference to a [`Mutator`] in a [`Mutate`] operator
+/// and use it in a chain of operators.
+impl<M, G> Mutator<G> for &M
+where
+    M: Mutator<G>,
+{
+    fn mutate(&self, genome: G, rng: &mut ThreadRng) -> anyhow::Result<G> {
+        (**self).mutate(genome, rng)
+    }
+}
+
+#[expect(
+    clippy::unwrap_used,
+    reason = "Panicking is the best way to deal with errors in unit tests"
+)]
+#[cfg(test)]
+mod tests {
+    use rand::{Rng, rngs::ThreadRng, thread_rng};
+
+    use super::Mutator;
+    use crate::operator::{Composable, Operator, mutator::Mutate};
+
+    struct FlipOne;
+
+    impl Mutator<Vec<bool>> for FlipOne {
```

Comment:
Also here do we use an actual Vector?

Reply:
I think we can get away with an array, although it does constrain all the tests to act on arrays of a fixed length (4 in this case). Not sure that matters, though.

Reply:
Then let's go for it :)